lib/components/WatchOnly.js

```diff
@@ -0,0 +1,70 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Text, Input, Button } from '@bpanel/bpanel-ui';
+import TerminalView from './TerminalView';
+
+class WatchOnly extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newWalletName: '',
+      xpub: '',
+    };
+  }
+
+  static get propTypes() {
+    return {
+      createWallet: PropTypes.func,
+    };
+  }
+
+  async input(key, value) {
+    await this.setState(
+      Object.assign({}, this.state, {
+        [key]: value,
+      })
+    );
+  }
+
+  create(walletId, xpub) {
+    const { createWallet } = this.props;
+
+    const opts = {
+      watchOnly: true,
+      accountKey: xpub,
+    };
+
+    createWallet(walletId, opts);
+  }
+
+  render() {
+    const { newWalletName, xpub } = this.state;
+
+    return (
+      <TerminalView>
+        <div>
+          <Text>Watch Only Wallet</Text>
```

Comment:
nit: should this be a header component maybe with type something like `h4`?

Reply:
updated